src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs

```diff
@@ -1,17 +1,125 @@
-        public IActionResult Index()
+        public async Task<IActionResult> Index(string employerAccountId)
         {
-            return View();
+            var accountId = _hashingService.DecodeValue(employerAccountId);
+
+            var reservationId = Guid.NewGuid();
+
+            await _mediator.Send(new CacheReservationEmployerCommand
+            {
+                Id = reservationId,
+                AccountId = accountId,
+                AccountLegalEntityId = 1,
+                AccountLegalEntityPublicHashedId = "111ABC",
+                AccountLegalEntityName = "Test Corp"
+            });
+
+            var viewModel = new ReservationViewModel{ Id = reservationId};
+
+            return View(viewModel);
+        }
+
+        [HttpGet]
+        [Route("{id}/SelectCourse",Name = RouteNames.EmployerSelectCourse)]
+        public async Task<IActionResult> SelectCourse(Guid id)
+        {
+            var getCoursesResponse = await _mediator.Send(new GetCoursesQuery());
+
+            var courseViewModels = getCoursesResponse.Courses.Select(c => new CourseViewModel(c));
+
+            var viewModel = new EmployerSelectCourseViewModel
+            {
+                ReservationId = id,
+                Courses = courseViewModels
+            };
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("{id}/SelectCourse")]
+        public async Task<IActionResult> PostSelectCourse(ReservationsRouteModel routeModel, string selectedCourseId)
+        {
+            var cachedReservation = await _mediator.Send(new GetCachedReservationQuery {Id = routeModel.Id.Value});
+
+            if (cachedReservation == null)
+            {
+                throw new ArgumentException("Reservation not found", nameof(routeModel.Id));
+            }
+
+            if (string.IsNullOrEmpty(selectedCourseId))
+            {
+                return RedirectToRoute(RouteNames.EmployerApprenticeshipTraining, new
+                {
+                    Id = cachedReservation.Id,
+                    EmployerAccountId = routeModel.EmployerAccountId,
+                });
+            }
+
+            try
+            {
+                await _mediator.Send(new CacheReservationCourseCommand
+                {
+                    Id = cachedReservation.Id,
+                    CourseId = selectedCourseId
+                });
+
+                return RedirectToRoute(RouteNames.EmployerApprenticeshipTraining, new
+                {
+                    Id = cachedReservation.Id,
+                    EmployerAccountId = routeModel.EmployerAccountId,
+                });
+            }
+            catch (ValidationException e)
+            {
+                foreach (var member in e.ValidationResult.MemberNames)
+                {
+                    ModelState.AddModelError(member.Split('|')[0], member.Split('|')[1]);
+                }
+
+                var getCoursesResponse = await _mediator.Send(new GetCoursesQuery());
```

Comment:
Might need to add some extra handling around this for if the courses dont come back